src/ui/pages/SwitchEthereumChain/SwitchEthereumChain.tsx

```diff
@@ -43,56 +50,100 @@ export function SwitchEthereumChain() {
-        <ZerionSquircle style={{ width: 44, height: 44 }} />
-        <Spacer height={16} />
-        <UIText kind="headline/h2" style={{ textAlign: 'center' }}>
-          Switch Chain Request ({chainId})
-        </UIText>
-        <Spacer height={8} />
-        <UIText kind="small/regular" color="var(--primary)">
-          {originName}
-        </UIText>
-        <Spacer height={8} />
-        <NetworkIndicator
-          chain={networks.getChainById(chainId)}
-          networks={networks}
-        />
-      </div>
-      <Spacer height={24} />
-      <UIText kind="small/regular" style={{ textAlign: 'center' }}>
-        <i>Screen to be completed</i>
-      </UIText>
-      <Spacer height={16} />
-      <Spacer height={16} />
+  const chain = networks.getChainById(chainId);
+  const network = networks.getNetworkByName(chain);
 
-      <VStack
-        style={{ textAlign: 'center', marginTop: 'auto', paddingBottom: 32 }}
-        gap={8}
-      >
-        <Button
-          onClick={() => {
-            const windowId = params.get('windowId');
-            invariant(windowId, 'windowId get-parameter is required');
-            windowPort.confirm(windowId, null);
+  return (
+    <Background backgroundKind="white">
+      <PageColumn>
+        <NavigationTitle title={null} documentTitle="Switch network" />
+        <PageTop />
+        <div
+          style={{
+            padding: 16,
+            border: '1px solid var(--neutral-400)',
+            borderRadius: 12,
           }}
         >
-          Approve
-        </Button>
-        <UnstyledButton
-          style={{ color: 'var(--primary)' }}
-          onClick={() => {
-            const windowId = params.get('windowId');
-            invariant(windowId, 'windowId get-parameter is required');
-            windowPort.reject(windowId);
+          <VStack gap={8}>
+            <HStack gap={8} alignItems="center">
+              <SiteFaviconImg url={origin} alt="" size={32} />
+              <UIText kind="headline/h2">{originName}</UIText>
+            </HStack>
+            <UIText kind="small/accent" color="var(--neutral-500)">
+              Suggests you switch current network
+            </UIText>
+          </VStack>
+        </div>
+        <Spacer height={24} />
+        {network ? (
+          <VStack gap={32} style={{ justifyItems: 'center' }}>
+            <VStack gap={4} style={{ justifyItems: 'center' }}>
+              <NetworkIcon
+                src={network.icon_url}
+                chainId={network.external_id || ''}
+                size={40}
+                name={network.name || null}
+              />
+              <UIText kind="headline/h1">{networks.getChainName(chain)}</UIText>
+            </VStack>
+            <VStack
+              gap={8}
+              style={{ justifyItems: 'center', textAlign: 'center' }}
+            >
+              <ValueCell
+                label="RPC URL"
+                value={network.rpc_url_public?.[0] || noValueDash}
```

Comment:
maybe
```suggestion
                value={networks.getRpcUrlPublic(chain)}
```

Reply:
I need to check that this will not cause extra errors. Btw, this line was copied from `src/ui/pages/Networks/NetworkCreateSuccess/NetworkCreateSuccess.tsx`